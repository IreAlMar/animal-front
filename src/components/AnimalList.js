import React from 'react';
import PropTypes from 'prop-types';
import AnimalListItem from './AnimalListItem';

const AnimalList = ({
  style,
  favorites,
  animals,
  ...props
}) => (
  <ul className="list-reset" style={style}>
    {animals.map(animal => (
      <AnimalListItem
        key={animal.taxonid}
        animal={animal}
        favorited={favorites.includes(animal.taxonid)}
        {...props}
      />
    ))}
  </ul>
);

AnimalList.propTypes = {
  animals: PropTypes.array,
  style: PropTypes.object,
  favorites: PropTypes.array,
};

export default AnimalList;
