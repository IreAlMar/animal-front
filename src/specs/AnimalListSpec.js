import React from 'react';
import renderer from 'react-test-renderer';
import AnimalList from '../components/AnimalList';

const testAnimals = [
  { taxonid: 1, scientific_name: 'Calumma fallax' },
  { taxonid: 2, scientific_name: 'Calumma vatosoa' },
  { taxonid: 3, scientific_name: 'Chamaeleo necasi' },
];

describe('<AnimalList />', () => {
  test('Should not break when no animals passed', () => {
    const component = renderer.create(<AnimalList />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly', () => {
    const component = renderer.create(<AnimalList animals={testAnimals} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorite state correctly', () => {
    const component = renderer.create(<AnimalList animals={testAnimals} favorites={[1]} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
