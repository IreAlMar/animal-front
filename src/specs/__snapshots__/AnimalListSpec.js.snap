// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimalList /> Should not break when no animals passed 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<AnimalList /> Should render correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ☐
    </span>
    <span>
      Calumma fallax
    </span>
    <span />
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ☐
    </span>
    <span>
      Calumma vatosoa
    </span>
    <span />
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ☐
    </span>
    <span>
      Chamaeleo necasi
    </span>
    <span />
  </li>
</ul>
`;

exports[`<AnimalList /> Should render favorite state correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ✅
    </span>
    <span>
      Calumma fallax
    </span>
    <span />
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ☐
    </span>
    <span>
      Calumma vatosoa
    </span>
    <span />
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ☐
    </span>
    <span>
      Chamaeleo necasi
    </span>
    <span />
  </li>
</ul>
`;
